pkg/crypto/validator.go

```diff
@@ -8,30 +8,24 @@ import (
 	"net"
 )
 
-type DefaultTlsValidator struct {
-	host string
-	port string
-}
+type DefaultTlsValidator struct{}
 
 type TlsValidator interface {
-	ValidateCert(cert string) error
-	HasSelfSignedCert() (bool, error)
+	ValidateCert(host, port, cert string) error
+	HasSelfSignedCert(host, port string) (bool, error)
 }
 
-func NewTlsValidator(host string, port string) TlsValidator {
-	return &DefaultTlsValidator{
-		host: host,
-		port: port,
-	}
+func NewTlsValidator() TlsValidator {
+	return &DefaultTlsValidator{}
 }
 
 // HasSelfSignedCert determines whether the url is using self-signed certs or not
-func (tv *DefaultTlsValidator) HasSelfSignedCert() (bool, error) {
+func (tv *DefaultTlsValidator) HasSelfSignedCert(host, port string) (bool, error) {
 	conf := &tls.Config{
 		InsecureSkipVerify: false,
 	}
 
-	_, err := tls.Dial("tcp", net.JoinHostPort(tv.host, tv.port), conf)
+	_, err := tls.Dial("tcp", net.JoinHostPort(host, port), conf)
 	if err != nil {
 		// If the error is x509.UnknownAuthorityError, this means the url is using self-signed certs
```

Comment:
I know this isn't the point of this PR, but I don't think this is accurate. Generally a certificate is self signed if the issuer and subject are the same but I think there are more complicated semantics to self-signed certs so using `x509` lib would be good.

We ought to verify its self signed by: (1) connect to acquire the cert ignoring any cert verification errors; (2) add the cert to a `CertPool`; (3) Use the `CertPool` as an option to `Certifiate.Verify()`.

See https://stackoverflow.com/a/63317898

Reply:
Makes sense. To be honest, I just moved the the struct members to method arguments to it's easier to use. I didn't stop to look at the code.

Should we create an issue for this and put it in the pipeline? Maybe @abhinavmpandey08 would have more context about this

Reply:
Yea I think a GH issue would be good.

Reply:
Of course there's the possibility that this method isn't named appropriately and its actual name is more akin to `IsAuthorityKnown()`.

Reply:
@chrisdoherty4 feel free to add/change anything https://github.com/aws/eks-anywhere/issues/1368